### PR TITLE
Main welcome

### PR DESCRIPTION
**Pull request by srishtigautam21**
Add:weather api unsplash api
Files to check :
mainPage.jsx
welcomePage.jsx
time.jsx
weather.jsx
todo.jsx
App.jsx

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
Code looks good to be merged

File: src/component/Greetings.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  const [greeting, setGreeting] = useState("");
+
+  useEffect(() => {
+    const hour = new Date().getHours();
+    if (hour < 4) setGreeting("Good Night");
+    else if (hour < 11) setGreeting("Good Morning");
+    else if (hour < 16) setGreeting("Good Afternoon");
+    else if (hour < 19) setGreeting("Good Evening");
+    else setGreeting("Good Night");
+  }, []);
```
Comment (itsmebipul00) [inline src/component/Greetings.jsx:13]: Appreciation: Nicely done

File: src/component/Todo/todo.css [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  cursor: pointer;
+  font-size: 2rem;
+  overflow-wrap: break-word;
+  color: var(--COLOR-WHITE);
+}
+
+.list-display input {
+  /* --size: 1.75rem;
+  width: var(--size);
+  height: var(--size); */
```
Comment (itsmebipul00) [inline src/component/Todo/todo.css]: Suggestion: You could remove the commented code

File: src/component/Weather/Weather.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  const [weather, setWeather] = useState(null);
+  const [city, setCity] = useState(localStorage.getItem("city"));
+  const [isInputVisible, setIsInputVisible] = useState(false);
+
+  const getWeatherData = async () => {
+    try {
+      const response =
+        coordinates.latitude !== 0 &&
+        (await axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?${
```
Comment (itsmebipul00) [inline src/component/Weather/Weather.jsx:16]: Suggestion: You could keep the APIs in a separate file

File: src/component/Weather/Weather.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  useEffect(() => {
+    getWeatherData();
+    // eslint-disable-next-line
+  }, [coordinates, city]);
+
+  return (
+    <div
+      className='weather-wrapper'
+      onKeyDown={(e) => e.code === "Enter" && setIsInputVisible(false)}
```
Comment (itsmebipul00) [inline src/component/Weather/Weather.jsx:60]: Appreciation: Nicely done

**Review by itsmebipul00 — APPROVED** [review]
Code is amazingly written. Good to merge Srishti

File: src/component/Greetings.jsx (hunk 1 above)
Comment (srishtigautam21) [reply src/component/Greetings.jsx:13]: thanks got this idea from sudip

File: src/component/Weather/Weather.jsx (hunk 4 above)
Comment (srishtigautam21) [reply src/component/Weather/Weather.jsx:60]: thanks

File: src/component/Weather/Weather.jsx (hunk 3 above)
Comment (srishtigautam21) [reply src/component/Weather/Weather.jsx:16]: ok will try to do this next pr

File: src/component/Todo/todo.css (hunk 2 above)
Comment (srishtigautam21) [reply src/component/Todo/todo.css]: done

**Review by dev-prabhat — APPROVED** [review]
Looks good to me 
Good to merge

**Pull request merged**